librz/core/canalysis.c

```diff
@@ -5755,6 +5755,98 @@ RZ_IPI bool rz_core_analysis_var_rename(RzCore *core, const char *name, const ch
 	return true;
 }
 
+static bool is_in_data_section(RzCore *core, const ut64 address) {
+	if (address < 256 || address == UT64_MAX) {
+		return false;
+	}
+	RzBinObject *o = rz_bin_cur_object(core->bin);
+	const RzBinSection *section = o ? rz_bin_get_section_at(o, address, core->io->va) : NULL;
+	return section && section->vsize > 0 && strstr(section->name, "data");
+}
+
+/**
+ * \brief      Tries to resolve all the constant pointers and adds flags named ptr.XXXXXX
+ *
+ * \param      core  The RzCore to analyze
+ */
+RZ_IPI void rz_core_analysis_resolve_pointers_to_data(RzCore *core) {
+	if (rz_str_startswith(rz_config_get(core->config, "asm.arch"), "dalvik")) {
+		// never run this for dalvik
+		return;
+	}
+
```

Comment:
why?

Reply:
there is no need to do this with dalvik stuff since everything is resolved by default. there are no pointers in memory.

Reply:
Add this as a comment to the code please

Reply:
The same might be true for other plugins as well? I don't like having special case in generic code like this, it feels hacky. Would anything happen if we just do it anyway? We should consider that someone not in rizinorg might write a plugin that we are not aware about and the non-plugin-specific code should work well nonetheless.

Reply:
i think many of these things can be solved once we have RzArch.

Reply:
How? I don't think It will be the magic solution for all of these special if we have around the code base

Reply:
just some options to say if the arch does use pointers etc.. like wasm does not either. it is a stack only based arch and the stack does not have a size. just a limit to how many values can be stored.

Reply:
Can we do that already in reanalysis plugins please? It's easier to port to rzarch if it's already in the interface rather than having to chase all ifs around

Reply:
what analysis plugin?